crates/zksync_error/Cargo.toml

```diff
@@ -0,0 +1,14 @@
+
+[package]
+name = "zksync_error"
+version = "0.1.0"
+edition = "2021"
+[lib]
+
+[dependencies]
+lazy_static = "1.5.0"
+serde = { version = "1.0.210", features = [ "derive", "rc" ] }
+serde_json = { version = "1.0.128" }
+strum = "0.26.3"
+strum_macros = "0.26.4"
+zksync-error-description = { git = "https://github.com/sayon/error-codegen-poc", branch = "main"}
```

Comment:
Last time I checked out your repo there was a way to generate code from `build.rs`, any reason not to use that here? From what I remember you were working on a way to reference dependencies via cargo thus not needing to use CLI at all. Admittedly I am a bit out of loop so sorry if this has been went through already :sweat_smile:

Reply:
We can still do it using `build.rs`! We can do it here too, it's just optional and a bit dirty (`build.rs` in one crate is used to generate code for another crate, as `build.rs` can not live in the workspace root...)

We can also fetch parts of error hierarchy from cargo packages rather then by direct links. However we need a language agnostic approach (or at least something that combines well with both JS and Rust ecosystems), and cargo is not :(

Reply:
Hmm, I was kind of expecting crate's `build.rs` to just write files to `./gen` and then we would `include!(../gen/mod.rs)` them from `src/lib.rs`. This is the most common approach I have seen in Rust ecosystem. Is there a reason this doesn't work for us?

> We can also fetch parts of error hierarchy from cargo packages rather then by direct links. However we need a language agnostic approach (or at least something that combines well with both JS and Rust ecosystems), and cargo is not :(

Makes sense

Reply:
I was the one who voted against `build.rs` approach -- while certainly neat, it makes it super hard to see the actual code being produced, which may kill the adoption (as not everyone is a rust expert). We can change the approach later, but for now I think that we should opt for ease of use.

Reply:
Well, we don't have to obfuscate the code generated from `build.rs` - I would pipe it through rustfmt and organize into proper files just like in this PR. And I would still keep them in the source control to see what actually gets changed in each PR that modifies error files. IMO typical user would observe only one difference between the two approaches: rust-analyzer/IDE support for `include!()` can be wonky at times.

Regardless, no strong opinion here from my side especially since I am super late to the party 😅